lib/commons/text/accessible-text.js

```diff
@@ -137,34 +138,34 @@ function nonEmptyText(t) {
  * @return {string}
  */
 text.accessibleText = function(element, inLabelledByContext) {
-	//todo: implement shadowDOM
-	var accessibleNameComputation;
-	var encounteredNodes = [];
+	let accessibleNameComputation;
+	const encounteredNodes = [];
+	if (element instanceof Node) {
+		element = axe.utils.getNodeFromTree(axe._tree[0], element);
+	}
 
 	function getInnerText (element, inLabelledByContext, inControlContext) {
-		var nodes = element.childNodes;
-		var returnText = '';
-		var node;
-
-		for (var i = 0; i < nodes.length; i++) {
-			node = nodes[i];
-			if (node.nodeType === 3) {
-				returnText += node.textContent;
-			} else if (node.nodeType === 1) {
-				if (phrasingElements.indexOf(node.nodeName.toUpperCase()) === -1) {
+		return element.children.reduce((returnText, child) => {
+			const { actualNode } = child;
+			if (actualNode.nodeType === 3) {
+				returnText += actualNode.nodeValue;
+			} else if (actualNode.nodeType === 1) {
+				if (!phrasingElements.includes(actualNode.nodeName.toUpperCase())) {
 					returnText += ' ';
 				}
-				returnText += accessibleNameComputation(nodes[i], inLabelledByContext, inControlContext);
+				returnText += accessibleNameComputation(
```

Comment:
Ugh... It's really frustrating going between axe-core and axe-devtools. :/ sorry